### PR TITLE
BufferedReader feature

### PR DESCRIPTION
**Pull request by peske**
I've extracted `BufferedReader` feature from my previous PR (https://github.com/felixse/FluentTerminal/pull/635), in order to make a simpler, separate PR.

@felixse, although you can merge this PR (`BufferedReader` class will be used anyway), it looks that it isn't enough. Please compare results of `cat text_file.txt` (where `text_file.txt` is a 5 megs text file):

- Current code (without any new PR): about 60 seconds;
- With `BufferedReader` implemented (this PR): about 30 seconds;
- With `BufferedReader`, protobuf and ZeroMQ (https://github.com/felixse/FluentTerminal/pull/635): about 9 seconds.

Obviously we should replace the serialization / communication mechanism.

### CODE REVIEW

File: FluentTerminal.SystemTray/BufferedRead.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace FluentTerminal.SystemTray
+{
+    internal sealed class BufferedReader : IDisposable
```
Comment (felixse) [inline FluentTerminal.SystemTray/BufferedRead.cs]: rename file to match class name

File: FluentTerminal.SystemTray/BufferedRead.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            _stream = stream;
+            _callback = callback;
+
+            // ReSharper disable once AssignmentIsFullyDiscarded
+            _ = Task.Factory.StartNew(ReadingLoop, TaskCreationOptions.LongRunning);
+        }
+
+        internal void SetPaused(bool value)
+        {
+            lock (_lock)
```
Comment (felixse) [inline FluentTerminal.SystemTray/BufferedRead.cs]: add some parenthesis or write it in one line

File: FluentTerminal.SystemTray/BufferedRead.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                _paused = value;
+        }
+
+        public void Dispose()
+        {
+            lock (_lock)
```
Comment (felixse) [inline FluentTerminal.SystemTray/BufferedRead.cs]: add some parenthesis or write it in one line

File: FluentTerminal.SystemTray/Services/ConPty/ConPtySession.cs [hunk 4]
```diff
@@ -73,35 +68,11 @@ private string GetWorkingDirectory(ShellProfile configuration)
 
         private void _terminal_OutputReady(object sender, EventArgs e)
         {
-            ListenToStdOut();
-        }
+            if (_reader != null)
```
Comment (felixse) [inline FluentTerminal.SystemTray/Services/ConPty/ConPtySession.cs]: add some parenthesis or write it in one line

**Comment by peske** [comment]
I've made it as you like it! :)

**Pull request merged**